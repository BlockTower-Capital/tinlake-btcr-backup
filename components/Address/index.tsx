import { Clone } from 'grommet-icons';
import copyToClipBoard from '../../utils/copyToClipBoard';
import { formatAddress } from '../../utils/formatAddress';
import styled from 'styled-components';

interface Props {
  address: string;
  [key: string]: any;
}

function Address({ address, ...rest }: Props) {
  return <AddressContainer onClick={() => copyToClipBoard(address)}
    title={address} {...rest}>
    {formatAddress(address)}{' '}
    <Clone size={'small'} />
  </AddressContainer>;
}

// tslint:disable-next-line:variable-name
const AddressContainer = styled.a`
  cursor: copy;
`;

export default Address;
