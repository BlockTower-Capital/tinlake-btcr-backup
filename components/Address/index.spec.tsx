import React from 'react';
import Address from '.';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<Address address="0x1234567891011" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
