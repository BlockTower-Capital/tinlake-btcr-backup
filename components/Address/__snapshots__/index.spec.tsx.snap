// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  className="sc-bdVaJa dyBADs"
  onClick={[Function]}
  title="0x1234567891011"
>
  0x1234...1011
   
  <svg
    aria-label="Clone"
    className="StyledIcon-ofa7kd-0 fjELgu"
    viewBox="0 0 24 24"
  >
    <path
      d="M7,23 L23,23 L23,7 L7,7 L7,23 Z M17,3.9997 L17,0.9997 L14,0.9997 M1,13.9997 L1,16.9997 L4,16.9997 M1,11.9997 L1,5.9997 L1,11.9997 Z M4,0.9997 L1,0.9997 L1,3.9997 M6,0.9997 L12,0.9997 L6,0.9997 Z"
      fill="none"
      stroke="#000"
      strokeWidth="2"
    />
  </svg>
</a>
`;
